pkg/odo/cli/service/operator_backend.go

```diff
@@ -121,17 +122,59 @@ func (b *OperatorBackend) ValidateServiceCreate(o *CreateOptions) (err error) {
 			// k8s does't have it installed by default but OCP does
 			return err
 		}
-
-		almExample, err := svc.GetAlmExample(csv, b.CustomResource, o.ServiceType)
+		b.group, b.version, b.resource, err = svc.GetGVRFromOperator(csv, b.CustomResource)
 		if err != nil {
 			return err
 		}
 
-		d.OriginalCRD = almExample
+		// if the service name is blank then we set it to custom resource name
+		if o.ServiceName == "" {
+			o.ServiceName = strings.ToLower(b.CustomResource)
+		}
 
-		b.group, b.version, b.resource, err = svc.GetGVRFromOperator(csv, b.CustomResource)
-		if err != nil {
-			return err
+		if len(o.parameters) != 0 {
+			var cr *olm.CRDDescription
+			hasCR := false
+			CRs := o.KClient.GetCustomResourcesFromCSV(&csv)
+			for _, custRes := range *CRs {
+				c := custRes
+				if c.Kind == b.CustomResource {
+					cr = &c
+					hasCR = true
+					break
+				}
+			}
+			if !hasCR {
+				return fmt.Errorf("the %q resource doesn't exist in specified %q operator", b.CustomResource, o.ServiceType)
+			}
+
+			crBuilder := service.NewCRBuilder(cr)
+			var errorStrs []string
+
+			for key, value := range o.ParametersMap {
+				err := crBuilder.SetAndValidate(key, value)
+				if err != nil {
+					errorStrs = append(errorStrs, err.Error())
+				}
+			}
+
+			if len(errorStrs) > 0 {
+				return errors.New(strings.Join(errorStrs, "\n"))
+			}
+
+			builtCRD, err := crBuilder.Map()
+			if err != nil {
+				return err
+			}
```

Comment:
Could you put this in a function, so we can more easily see the parallel with svc.GetAlmExample?

Reply:
ack, good suggestion